### DISCUSSION TITLE
how to calculate cost when sending NFT + datum

### DISCUSSION THREAD

**Discussion by reitcircles** [Python-Cardano/pycardano#304] [General]
`min_val = min_lovelace(
    chain_context, output=TransactionOutput(address, Value(0, my_nft))
)`

How do i calculate the cost when a datum is associated with the nft (basically sending it to a Smart contract address) using the min_lovelace function ?

**Comment by nielstron** [comment]
Did you try adding the `datum` parameter to the TransactionOutput?

I guess you are talking about minUtxo and not transaction fee (the latter is automatically calculated when building a transaction in the transaction builder)

**Reply by reitcircles** [reply]
yes. i tried datum=true, but it did not change the output. FInally i just calculated the number from the chain params. That worked.

**Reply by nielstron** [reply]
The datum parameter needs to be set to the actual datum that you attach, not true. Basically you pass the entire transactionoutput the way you would submit it on-chain into the function and then just have to adjust the Lovelace amount in the output.
